lib/stringify.js

```diff
@@ -235,7 +233,7 @@ var JSON = module.exports;
 
         switch (typeof value) {
         case 'string':
-            if (isBigNumber) {
+            if (isNumber) {
```

Comment:
How can something of type `string` can have isNumber = true?

Also, we have this code above:

```
        if (value && typeof value === 'object' &&
                typeof value.toJSON === 'function') {
            value = value.toJSON(key);
        }
```

It looks like a leftover - do we rely on it in our logic? What if we throw it away?

Reply:
These are both direct lifts from the original code. I also wasn't sure why either of them is there: happy to remove.

Reply:
In particular, that code has a comment explaining what it does:

```
// If the value has a toJSON method, call it to obtain a replacement value.
```

I guess it's not _massively_ important for us, but I don't think it's that bad a thing to leave in.